### PR TITLE
Add developer guide 

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
Add developer guide to maps-dashboards plugin.

### Issues Resolved
Closes https://github.com/opensearch-project/dashboards-maps/issues/105

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: maps_dashboards/ DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+1. Download the OpenSearch Dashboards source code for the version specified in package.json you want to set up.
+2. Change your node version by `nvm use <version>`to the version specified in `.node-version` inside the OpenSearch Dashboards root directory.
+3. Create a `plugins` directory inside the OpenSearch Dashboards source code directory, if `plugins` directory doesn't exist.
+4. cd into `plugins` directory in the OpenSearch Dashboards source code directory.
+5. Check out this package from version control into the `plugins` directory.
+```
+git init
+// Change to your forked repo link in below command
+git remote add -f origin https://github.com/opensearch-project/dashboards-maps.git
```
Comment (VijayanB) [inline maps_dashboards/ DEVELOPER_GUIDE.md:32]: I think we need set up for both plugins. Shall we add something like
to checkout maps:
... 
...
...

to check out custom import
...
...
..

Comment (junqiu-lei) [reply maps_dashboards/ DEVELOPER_GUIDE.md:32]: For development, we don't need to check out all the two plugins. The sparse-checkout will help only check out `/maps-dashboards` folder, so we only need to set up maps-dashboards plugin.

File: maps_dashboards/ DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+### Submitting Changes
+
+See [CONTRIBUTING](CONTRIBUTING.md).
+
+### Backports
+
+The Github backport workflow creates backport PRs automatically when the original PR
+with an appropriate label `backport <backport-branch-name>` is merged to main with the backport workflow run successfully on the
+PR. For example, if a PR on main needs to be backported to `1.x` branch, add a label `backport 1.x` to the PR and make sure the
```
Comment (martin-gaievski) [inline maps_dashboards/ DEVELOPER_GUIDE.md:72]: I think first sentence is a bit too long, maybe split into two: 
```
The Github backport workflow creates backport PRs automatically for PRs with label `backport <backport-branch-name>`. Label should be attached to the original PR, backport workflow starts when original PR merged to main branch.

File: maps_dashboards/ DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+### Run
+
+From OpenSearch-Dashbaords repo (root folder), run the following command:
+
+```bash
+yarn osd bootstrap
+yarn start
+```
+
+  Starts OpenSearch Dashboards and includes this plugin. OpenSearch Dashboards will be available on `localhost:5601`.
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply maps_dashboards/ DEVELOPER_GUIDE.md]: updated

File: maps_dashboards/ DEVELOPER_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+### Test
+
+From maps-dashboards folder running the following command runs the plugin unit tests -
```
[the remark this answers is not in the document]
Comment (junqiu-lei) [reply maps_dashboards/ DEVELOPER_GUIDE.md]: updated

File: maps_dashboards/ DEVELOPER_GUIDE.md (hunk 2 above)
Comment (junqiu-lei) [reply maps_dashboards/ DEVELOPER_GUIDE.md]: Thanks, updated.

**Review by VijayanB — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]
looks good, thx

**Pull request merged**